cocotb/share/lib/verilator/verilator.cpp

```diff
@@ -84,5 +84,11 @@ int main(int argc, char** argv) {
     tfp->close();
 #endif
 
+// VM_COVERAGE is a define which is set if Verilator is 
+// instructed to collect coverage (when compiling the simulation)
+#if VM_COVERAGE
+    VerilatedCov::write("coverage.dat");
```

Comment:
Just to confirm, we want always ``coverage.dat`` here?

Reply:
The verilator docs explain that you want a different coverage.dat for each test run (and suggests putting them in logs/coverage.dat, FWIW) so that you can merge them all together after you run all your tests.

Does cocotb make it easy to run the different tests in different directories?  If so, then this will be fine.  If not then this is probably fine for now and we can figure out a better way to separate coverage.dat files in a separate commit (IMHO).

Reply:
Updated the branch as per suggestions. Regarding coverage file name, I would ideally like a name with <topmodule>_<test name>_<seed>.dat or to put coverage.dat in a test directory name like that. Not sure which helps for merging. 

But I think this would be okay for now.

Reply:
You can make it define that defaults `coverage.dat` and set with `COMPILE_ARGS` if needed.

Reply:
> Does cocotb make it easy to run the different tests in different directories?

Only via https://cocotb.readthedocs.io/en/latest/building.html#var-SIM_BUILD which is per cocotb invocation.